lib/src/execution/windows/execution.cc

```diff
@@ -195,11 +195,151 @@ namespace facter { namespace execution {
         return commandline;
     }
 
-    pair<bool, string> execute(
+    // Typedef for pipe data
+    typedef tuple<HANDLE, OVERLAPPED*, bool, string*, function<bool(string const&)> const*> pipe_data;
```

Comment:
Yeah, this kind of organically grew as more and more data was needed in the helper function :/  I'll take a look at using a struct to clean it up a little.